### ISSUE TITLE
Python's os.minor() does not support devices with high minor numbers

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#1114]
Originally reported of Google Code with ID 1058.

```
As outlined in Debian bug #782073 (https://bugs.debian.org/782073), Python's os.minor() does not support devices with high minor numbers. This is due to Python using an old definition of minor:

  def minor(dev): return ((int)((dev) & 0xff))

instead of glibc's newer one:

  return (__dev & 0xff) | ((unsigned int) (__dev >> 12) & ~0xff);

This causes extstorage to break on systems with multiple devices (which is not unlikely when using dm-multipath with FC/iSCSI SANs).

Note that major() is also affected by a similar issue, but it's highly unlikely to be encountered because the SCSI subsystem uses relatively low major numbers.

Although this is a bug in Python, I think it's best to provide our own minor() implementation working around this issue.
```
Originally added on 2015-04-07 12:50:44 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
I'll have a look.
```
Originally added on 2015-04-07 14:03:52 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit 7cbacba98a330c464e1cda67442da0ad43a833a2
Author: Klaus Aehlig <aehlig@google.com>
Date:   Tue Apr 7 18:06:13 2015 +0200

    Switch to our osminor
    
    ...and thus work around the bug in os.minor
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Hrvoje Ribicic <riba@google.com>

commit 01cf0021c814207da0200337772ba3e73ff0b178
Author: Klaus Aehlig <aehlig@google.com>
Date:   Tue Apr 7 18:04:28 2015 +0200

    Provide an alternative for os.minor working around its bug
    
    Python's os.minor still contains an old definition, whereas the
    current one has changed. So we add our own definition working around
    this bug.
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Hrvoje Ribicic <riba@google.com>

```
Originally added on 2015-04-08 09:42:07 +0000 UTC.

Added to Milestone: Release2.10
Removed from Milestone: Release2.13

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit 83d75a6bf6dd8de26aa6053f35cf6830eca109ac
Author: Klaus Aehlig <aehlig@google.com>
Date:   Wed Apr 8 16:51:48 2015 +0200

    Replace additional os.minor by our custom version
    
    ...thus avoiding the bug in the python library.
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Hrvoje Ribicic <riba@google.com>

```
Originally added on 2015-04-08 15:53:44 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
With that change, all occurrences of os.minor have been replaced by
a compatible version. Can someone please verify that this also checks
the problems that were observed in the original bug report?
```
Originally added on 2015-04-08 15:56:18 +0000 UTC.

Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Released as part of 2.14.0
```
Originally added on 2015-06-02 10:40:20 +0000 UTC.

Changed State: Released
